storage/engine/rocksdb/db.cc

```diff
@@ -324,6 +332,39 @@ class DBCompactionFilterFactory : public rocksdb::CompactionFilterFactory {
   int64_t min_txn_ts_;
 };
 
+class DBPrefixExtractor : public rocksdb::SliceTransform {
+ public:
+  DBPrefixExtractor() {
+  }
+
+  virtual const char* Name() const {
+    return "cockroach_prefix_extractor";
+  }
+
+  // MVCC keys are encoded as <user-key>/<timestamp>. Extract the <user-key>
+  // prefix which will allow for more efficient iteration over the keys
+  // matching a particular <user-key>. Specifically, the <user-key> will be
+  // added to the per table bloom filters and will this be able to skip tables
+  // which do not contain the <user-key>.
+  virtual rocksdb::Slice Transform(const rocksdb::Slice& src) const {
+    const char kTerm[] = "\x00\x01";
+    const int kTermSize = sizeof(kTerm) - 1;
+    const char* p = reinterpret_cast<const char*>(memmem(src.data(), src.size(), kTerm, kTermSize));
```

Comment:
Because keys for `MVCCMetadata` values do not contain the 12-byte timestamp suffix.